lib/omnibus/packagers/windows_base.rb

```diff
@@ -90,22 +90,84 @@ def thumbprint
       signing_identity[:thumbprint]
     end
 
-    def algorithm
-      signing_identity[:algorithm]
-    end
-
     def cert_store_name
       signing_identity[:store]
     end
 
-    def timestamp_servers
-      signing_identity[:timestamp_servers]
-    end
-
     def machine_store?
       signing_identity[:machine_store]
     end
 
+    def signing_identity_file(pfxfile = NULL, params = NULL)
+      unless null?(pfxfile)
+        @signing_identity_file = {}
```

Comment:
As a rule of thumb normally memoization isn't ideal when a method takes in parameters as the first execution is the one that will be "memoized", and future calls will always return the same one. Maybe in the context of this it would be fine (I see they used the same thing for `signing_identity`).... but just a thought.

Reply:
This is pretty much the pattern everywhere in omnibus.  I just duplicated `signing_identity`, but that's pretty much  how the whole repo works.  We could do it differently here if we thought it was necessary, but I just copy/pasted `signing_identity`

Reply:
I think that's kind of the "goal" of the method, and many similar methods on `omnibus-ruby` work the same way. It's a bizarre way to implement getters and setters:
- you pass a value to the method => sets the value of the "attribute" and returns it
- you don't pass a value to the method => gets the value of the "attribute"

Here, you'd use it like this:
 - in the software definition, you set it once like this:
`signing_identity_file "filename", option1: value1, option2: value2`
- and then it's used internally by omnibus ruby, only as an "attribute":
`if signing_identity_file`